### DISCUSSION TITLE
Show which list blocked DNS query

### DISCUSSION THREAD

**Discussion by DarthSonic** [celzero/rethink-app#636] [Ideas]
It would be very helpful if the log could show which list is blocking the DNS query. I know that DNS response is not returning such information but maybe you could add an option for self-hosted worker DNS resolvers as a fourth type of DNS resolvers, and then be able to analyze a block in the log by requesting this information from the self-hosted worker?

**Comment by ignoramous** [comment]
This is slightly complicated because the app doesn't know which `version` (which is nothing but a unix timestamp) the self-hosted Rethink DNS server is running. We can let the user set that, but prone to error, since it needs to be manually done each and every time a new blocklist `version` is available (which, for the Rethink DNS servers is *every* week). Another possibility is to assume that the self-hosted Rethink DNS server is running on the same `version` as the default Rethink DNS deployment... that would simplify things a bit. I will think a bit over this.
